Traceback (most recent call last):
  File "/scratch/07169/wgl/adjoint/python/Main_no_matrix.py", line 186, in <module>
    params, opt_state = trainStep(u_0_train, t, dt, params, net, opt_state,
  File "/scratch/07169/wgl/adjoint/python/Main_no_matrix.py", line 130, in trainStep
    vmap(grad(lossFn, argnums=(3,)),
  File "/work/07169/wgl/ls6/PHO-ICES/adj-env/lib/python3.9/site-packages/jax/_src/traceback_util.py", line 166, in reraise_with_filtered_traceback
    return fun(*args, **kwargs)
  File "/work/07169/wgl/ls6/PHO-ICES/adj-env/lib/python3.9/site-packages/jax/_src/api.py", line 1253, in vmap_f
    out_flat = batching.batch(
  File "/work/07169/wgl/ls6/PHO-ICES/adj-env/lib/python3.9/site-packages/jax/_src/linear_util.py", line 166, in call_wrapped
    ans = self.f(*args, **dict(self.params, **kwargs))
  File "/work/07169/wgl/ls6/PHO-ICES/adj-env/lib/python3.9/site-packages/jax/_src/traceback_util.py", line 166, in reraise_with_filtered_traceback
    return fun(*args, **kwargs)
  File "/work/07169/wgl/ls6/PHO-ICES/adj-env/lib/python3.9/site-packages/jax/_src/api.py", line 659, in grad_f
    _, g = value_and_grad_f(*args, **kwargs)
  File "/work/07169/wgl/ls6/PHO-ICES/adj-env/lib/python3.9/site-packages/jax/_src/traceback_util.py", line 166, in reraise_with_filtered_traceback
    return fun(*args, **kwargs)
  File "/work/07169/wgl/ls6/PHO-ICES/adj-env/lib/python3.9/site-packages/jax/_src/api.py", line 739, in value_and_grad_f
    _check_scalar(ans)
  File "/work/07169/wgl/ls6/PHO-ICES/adj-env/lib/python3.9/site-packages/jax/_src/api.py", line 759, in _check_scalar
    raise TypeError(msg(f"had shape: {aval.shape}"))
jax._src.traceback_util.UnfilteredStackTrace: TypeError: Gradient only defined for scalar-output functions. Output had shape: (1,).
The stack trace below excludes JAX-internal frames.
The preceding is the original exception that occurred, unmodified.
--------------------
The above exception was the direct cause of the following exception:
Traceback (most recent call last):
  File "/scratch/07169/wgl/adjoint/python/Main_no_matrix.py", line 186, in <module>
    params, opt_state = trainStep(u_0_train, t, dt, params, net, opt_state,
  File "/scratch/07169/wgl/adjoint/python/Main_no_matrix.py", line 130, in trainStep
    vmap(grad(lossFn, argnums=(3,)),
TypeError: Gradient only defined for scalar-output functions. Output had shape: (1,).